### PR TITLE
Make skill functions in player and character consistent 

### PR DESCRIPTION
**Pull request by BevapDin**
Instead of having a bunch of overloaded functions, only use the skill id all the time. This is consistent with other functions like `has_trait` or `has_effect` or `get_effect`.

It also moves `boost_skill_level` and `set_skill_level` into the `Character` class. The skill map and `get_skill_level` read already there.

Renames `Character::skillLevel` to `get_skill_level` for consistency.

Replaces direct changes of the skill level (via `get_skill_level(id).level(level)`) to `set_skill_level`.

Change the skill map to have string ids as keys, consistent with the other maps.

### CODE REVIEW

File: src/character.h [hunk 1]
```diff
@@ -402,14 +402,12 @@ class Character : public Creature, public visitable<Character>
         bool worn_with_flag( std::string flag ) const;
 
         // --------------- Skill Stuff ---------------
-        SkillLevel &skillLevel(const Skill* _skill);
-        SkillLevel &skillLevel(Skill const &_skill);
-        SkillLevel &skillLevel(const skill_id &ident);
+        SkillLevel &get_skill_level( const skill_id &ident );
```
Comment (Coolthulhu) [inline src/character.h]: Do we need that thing? Having the ability to set skill level field value directly only prevents creation of reliable "on levelup" triggers for things like learning Brawling.

File: src/player.cpp [hunk 2]
```diff
@@ -12327,65 +12328,65 @@ void player::practice( const Skill* s, int amount, int cap )
...
     if (amount > 0 && level.isTraining()) {
-        int oldLevel = get_skill_level(s);
-        skillLevel(s).train(amount);
-        int newLevel = get_skill_level(s);
+        int oldLevel = get_skill_level( id );
+        get_skill_level( id ).train(amount);
+        int newLevel = get_skill_level( id );
         if (is_player() && newLevel > oldLevel) {
-            add_msg(m_good, _("Your skill in %s has increased to %d!"), s->name().c_str(), newLevel);
+            add_msg(m_good, _("Your skill in %skill has increased to %d!"), skill.name().c_str(), newLevel);
```
Comment (Coolthulhu) [inline src/player.cpp]: Looks like too eager autoreplace here.

File: src/character.h (hunk 1 above)
Comment (BevapDin) [inline src/character.h]: It's currently still used to set a few properties of the `SkillLevel` object, e.g. train, rust, exercise, etc.

I though about (later) making the function protected, or accessing `Character::_skills` directly.

**Comment by Coolthulhu** [comment]
Looks like this PR somehow sneaked in a `t-static.s` file.

**Pull request merged**